_ingest-pipelines/processors/dissect.md

```diff
@@ -0,0 +1,289 @@
+
+The dissection algorithm is precise and requires that every character in the pattern exactly match the source string. For instance, the pattern `%{helloworldkey} %{worldkey}` (one space) will match the string "Hello world" (one space) but not the string "Hello  world" (two spaces) because pattern only has one space while the source string has two.
+
+The right padding modifier can be used to address this issue. By adding the right padding modifier to the pattern `%{helloworldkey->} %{worldkey}`, it will no match `Hello world` (one space), `Hello  world` (two spaces), and even `Hello          world` (ten spaces). 
+
+The right padding modifier is used to allow for the repetition of characters following a `%{keyname->}`. The right padding modifier can be applied to any key along with any other modifiers. It should always be the rightmost modifier, for example, `%{+keyname/1->}`, `%{}`.
+
+#### Example of usage
+
+The following is an example of how to use a right padding modifier:
+
+`%{name->} %{city}, %{state} %{zip}`
+
+In this pattern, the right padding modifier `->` is applied to the `%{name}` key. This means that the `%{name}` key will match an sequence of characters, including spaces. This is useful for handling names that may contain spaces, such as "First Last".
+
+The following is an example of how the right padding would be used to extract information from the following address entries: 
+
+```bash
+New York, NY 10017
+New York City, NY 10017
+```
+
+Both addresses contain the same information, but the second entry has an extra word, `City`, in the city field. The right padding modifier allows the pattern to match both of these address entries, even though they have slightly different formats.
+
+### Append modifier (`+`)
+
+The append modifier combines the values of two or more keys into a single output value. The values are appended from left to right. You can also specify an optional separator to be inserted between the values. 
+
+#### Example of usage
+
+The following is an example of how to use the append modifier. In this example, the pattern extracts the values of `key1` and `key2` fields and appends them together, with a space as the separator:
+
+`%{key1} %{key2}`
+
+The output is:
+
+`value1 value2`
+
+You can also specify a custom separator using the `append_separator` parameter. For example, the following pattern uses a comma as the separator: 
+
+`%{key1} %{key2}, append_separator => ","`
+
+The output is:
+
+`value1, value2`
+
+### Append with order modifier (`+` and `/`)
+
+The append with order modifier combines the values of two or more keys into a single output value based on the order specified after the `/`. You have the flexibility to customize the separator that separates the appended values. The append modifier is useful for compiling multiple fields into a single formatted output line, constructing structured lists of data items, and consolidating values from various sources.  
```

Comment:
This section does not show an order modifier. Here's an example that does:

Specify reverse order for the previous example
```
PUT /_ingest/pipeline/dissect-test
{
  "description": "Pipeline that dissects web server logs",
  "processors": [
    {
      "dissect": {
        "field": "message",
        "pattern": "%{+address/3}, %{+address/2} %{+address/1}",
        "append_separator": "|"
      }
    }
  ]
}
```
Test the pipeline with the same request:
```
POST _ingest/pipeline/dissect-test/_simulate
{
  "docs": [
    {
      "_index": "testindex1",
      "_id": "1",
      "_source": {
        "message": "New York, NY 10017"
      }
    }
  ]
}
```
The substrings are appended into the `address` fields in reverse order:
```
{
  "docs": [
    {
      "doc": {
        "_index": "testindex1",
        "_id": "1",
        "_source": {
          "address": "10017|NY|New York",
          "message": "New York, NY 10017"
        },
        "_ingest": {
          "timestamp": "2024-01-22T22:38:24.305974178Z"
        }
      }
    }
  ]
}
```